api/src/context/java/io/grpc/Deadline.java

```diff
@@ -95,6 +101,11 @@ public static Deadline after(long duration, TimeUnit units, Ticker ticker) {
     return new Deadline(ticker, units.toNanos(duration), true);
   }
 
+  public static Deadline after(Duration duration, Ticker ticker) {
+    return after(TimeUnit.NANOSECONDS.convert(duration.getSeconds(), TimeUnit.SECONDS),
```

Comment:
I have a concern about the loss of precision, I have raised a question on Eric's comment to use TimeUnit.NANOSECONDS.convert(duration) instead of Duration.toNano().

Reply:
As Eric replied, you should be using https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/TimeUnit.html#convert(java.time.Duration) and not have to convert using the seconds value.